docs/wave-intrinsics.md

```diff
@@ -5,6 +5,110 @@ Slang has support for Wave intrinsics introduced to HLSL in SM6.0 and SM6.5. All
 
 Another wrinkle in compatibility is that on GLSL targets such as Vulkan, the is not built in language support for Matrix versions of Wave intrinsics. Currently this means that Matrix is not a supported type for Wave intrinsics on Vulkan, but may be in the future.
 
+The Wave Intrinsics supported on Slang are listed below. Note that typically T generic types also include vector and matrix forms. 
+
+```
+
+// Lane info
+
+uint WaveGetLaneCount();
+
+uint WaveGetLaneIndex();
+
+bool WaveIsFirstLane();
+
+// Ballot
+
+bool WaveActiveAllTrue(bool condition);
+
+bool WaveActiveAnyTrue(bool condition);
+
+uint4 WaveActiveBallot(bool condition);
+
+uint WaveActiveCountBits(bool value);
+
+// Barriers 
+
+void AllMemoryBarrierWithWaveSync();
+
+void GroupMemoryBarrierWithWaveSync();
+
+// Across Lanes
+
+__generic<T : __BuiltinIntegerType>
```

Comment:
We probably ought to scrub these declarations so that they don't use `__generic` in our docs:

```
T WaveActiveBitAnd<T>(T expr);
```

I'm on the fence about whether we should just write `<T>` (which is a little bit misleading) or `<T : __BuiltinIntegerType>`, which is accurate but exposes the `__BuiltinIntegerType` interface which is currently not really intended for the eyes of users. It seems like it might be best to go with `<T : __BuiltinIntegerType>` but to let this light a fire under us to rename those built-in `interface`s to something we are ready to have users see and think about.